Bot.Core/GhostTownCleanup.cs

```diff
@@ -0,0 +1,56 @@
+﻿using Bot.Api;
+using Bot.Api.Database;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Bot.Core
+{
+    internal class GhostTownCleanup
+    {
+        ITownMaintenance m_startupTownTasks;
+        IBotClient m_botClient;
+        ITownDatabase m_townDatabase;
+        IGameMetricDatabase m_gameMetricDatabase;
+        IDateTime m_dateTime;
+
+        public GhostTownCleanup(IServiceProvider sp)
+        {
+            sp.Inject(out m_startupTownTasks);
+            sp.Inject(out m_botClient);
+            sp.Inject(out m_townDatabase);
+            sp.Inject(out m_gameMetricDatabase);
+            sp.Inject(out m_dateTime);
+
+            m_startupTownTasks.AddMaintenanceTask(CleanupGhostTown);
+        }
+
+        private async Task CleanupGhostTown(TownKey townKey)
+        {
+            Serilog.Log.Information("GhostTownCleanup: Checking town {townKey}", townKey);
+
+            // If there was activity in the last 30 days, skip this town
+            var mostRecent = await m_gameMetricDatabase.GetMostRecentGame(townKey);
+            if (mostRecent != null && mostRecent.Value.AddDays(30) > m_dateTime.Now)
```

Comment:
What happens if this is a brand-new town that was just created? Does it have an entry in the database? Might this accidentally kill a town that just got created a few hours ago?

Reply:
Yes, it could, but only if the Guild or its control channel were not able to be resolved 🤔